backend/src/main/java/ai/verta/modeldb/experimentRun/FutureExperimentRunDAO.java

```diff
@@ -874,4 +877,30 @@ public FutureExperimentRunDAO(
             ModelDBActionEnum.ModelDBServiceActions.UPDATE)
         .thenCompose(unused -> createExperimentRunHandler.createExperimentRun(request), executor);
   }
+
+  public InternalFuture<Void> logEnvironment(LogEnvironment request) {
+    final var runId = request.getId();
+
+    if (!request.hasEnvironment()) {
+      return InternalFuture.failedStage(
+          new InvalidArgumentException("Environment should not be empty"));
+    }
+
+    return checkPermission(
+            Collections.singletonList(runId), ModelDBActionEnum.ModelDBServiceActions.READ)
+        .thenCompose(
+            unused ->
+                jdbi.useHandle(
+                    handle ->
+                        handle
+                            .createUpdate(
+                                "UPDATE experiment_run SET environment = :environment WHERE id = :runId")
+                            .bind("runId", request.getId())
+                            .bind(
+                                "environment",
+                                environmentHandler.getEnvironmentStringFromBlob(
+                                    request.getEnvironment()))
+                            .execute()),
```

Comment:
Why is this not in the environment handler?

Reply:
Moved to environment handler